packages/react-native/ReactAndroid/src/test/java/com/facebook/react/packagerconnection/JSPackagerClientTest.kt

```diff
@@ -0,0 +1,137 @@
+package com.facebook.react.packagerconnection
+
+import com.facebook.react.packagerconnection.ReconnectingWebSocket.ConnectionCallback
+import java.io.IOException
+import okio.ByteString
+import okio.ByteString.Companion.encodeUtf8
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.*
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class JSPackagerClientTest {
+  private fun createRH(action: String, handler: RequestHandler): Map<String, RequestHandler> {
+    val m: MutableMap<String, RequestHandler> = HashMap()
+    m[action] = handler
+    return m
+  }
+
+  private lateinit var mSettings: PackagerConnectionSettings
+
+  @Before
+  fun setup() {
+    mSettings = mock(PackagerConnectionSettings::class.java)
+    `when`(mSettings.debugServerHost).thenReturn("ws://not_needed")
+    `when`(mSettings.packageName).thenReturn("my_test_package")
+  }
+
+  @Test
+  @Throws(IOException::class)
+  fun test_onMessage_ShouldTriggerNotification() {
+    val handler = mock(RequestHandler::class.java)
+    val client = JSPackagerClient("test_client", mSettings, createRH("methodValue", handler))
+
+    client.onMessage(
+      "{\"version\": 2, \"method\": \"methodValue\", \"params\": \"paramsValue\"}"
+    )
+    verify(handler).onNotification(eq("paramsValue"))
+    verify(handler, never()).onRequest(any(), any(Responder::class.java))
+  }
+
+  @Test
+  @Throws(IOException::class)
+  fun test_onMessage_ShouldTriggerRequest() {
+    val handler = mock(RequestHandler::class.java)
+    val client = JSPackagerClient("test_client", mSettings, createRH("methodValue", handler))
+
+    client.onMessage(
+      "{\"version\": 2, \"id\": \"idValue\", \"method\": \"methodValue\", \"params\": \"paramsValue\"}"
+    )
+    verify(handler, never()).onNotification(any())
+    verify(handler).onRequest(eq("paramsValue"), any(Responder::class.java))
+  }
+
+  @Test
+  @Throws(IOException::class)
+  fun test_onMessage_WithoutParams_ShouldTriggerNotification() {
+    val handler = mock(RequestHandler::class.java)
+    val client = JSPackagerClient("test_client", mSettings, createRH("methodValue", handler))
+
+    client.onMessage("{\"version\": 2, \"method\": \"methodValue\"}")
+    verify(handler).onNotification(eq<Any?>(null))
```

Comment:
```suggestion
    verify(handler).onNotification(ArgumentMatchers.isNull())
```

This should also work